pkg/api/build.go

```diff
@@ -40,3 +41,29 @@ func (api Build) Jobs(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, jobs)
 }
+
+// Logs creates a new gin handler for the GET /job/:id/logs endpoint
+func (api Build) Logs(c *gin.Context) {
+	id := c.Params.ByName("id")
+	build, err := api.store.GetBuild(id)
+	if err != nil {
+		c.JSON(http.StatusNotFound, struct{}{})
+		return
+	}
+	if c.Query("stream") == "true" {
+		logReader, err := api.store.GetWorkerLogStream(build.Worker)
+		if err != nil {
+			c.JSON(http.StatusNotFound, struct{}{})
+			return
+		}
+		defer logReader.Close()
+		io.Copy(c.Writer, logReader)
+	} else {
+		logs, err := api.store.GetWorkerLog(build.Worker)
+		if err != nil {
+			c.JSON(http.StatusNotFound, struct{}{})
+			return
+		}
+		c.JSON(http.StatusOK, logs)
```

Comment:
Should we check for a length of zero and return a 204 No Content when there's no logs to ship back? Might make it a bit more user-friendly.

Reply:
/cc @seanknox Would a 204 return code help or hurt? In this implementation, it will actually send back a `""\n`, which is a JSON-encoded empty string.

Reply:
204 return code would be good to do as it'll make the client parsing code in kashti clearer (e.g. `if responseCode == 204` vs `if response.body == null`).